src/window.rs

```diff
@@ -59,3 +63,152 @@ pub fn settings() -> Settings {
         },
     }
 }
+
+pub fn events() -> Subscription<Event> {
+    Subscription::from_recipe(Events)
+}
+
+enum State<T: Stream<Item = Event>> {
+    Idle {
+        stream: T,
+    },
+    Moving {
+        stream: T,
+        position: window::Position,
+    },
+    Resizing {
+        stream: T,
+        size: window::Size,
+    },
+    MovingAndResizing {
+        stream: T,
+        position: window::Position,
+        size: window::Size,
+    },
+}
+
+struct Events;
+
+impl subscription::Recipe for Events {
+    type Output = Event;
+
+    fn hash(&self, state: &mut subscription::Hasher) {
+        use std::hash::Hash;
+
+        std::any::TypeId::of::<Self>().hash(state);
+    }
+
+    fn stream(self: Box<Self>, events: subscription::EventStream) -> BoxStream<'static, Self::Output> {
+        use futures::stream;
+        const TIMEOUT: std::time::Duration = std::time::Duration::from_secs(3);
```

Comment:
3 seconds feels a bit long? For instance, a double click has to register typically within 500ms. Maybe 1 second?

Reply:
Changing to 500ms.